### ISSUE TITLE
[feature request] Save mindmap as image

### ISSUE THREAD

**Issue by lgg** [tobloef/text2mindmap#6]
Please, can you add ability to export mindmap as jpg/png, maybe even SVG.

**Comment by tobloef** [comment]
Hi there.

This is a good suggestion and it's something I'm working on as part of the rewrite. You can read more about it in the recently updated README. 

Stay tuned.

**Comment by cavo789** [comment]
Nice suggestion. Indeed, when exporting the HTML to f.i. docx or pdf, saving the mindmap to an image will be usefull

**Comment by arendtio** [comment]
@lgg you did notice that the mindmap is rendered on a canvas, so simply doing a right click and selecting 'save graphic as' works in Firefox (and probably some other browsers) already?

**Comment by cavo789** [comment]
@arendtio Right ! Tested just right now with Chrome and it's possible to save the map as a png file.

**Comment by thib92** [comment]
@tobloef Canvas has a very useful method in JS called `toDataUrl()`, which allows you to export the content of the canvas as an image. [Read more](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL)

**Comment by joserbngithub** [comment]
Absolutely. I'm using your page and it's great, it's already behaving like the old txt2mindmap. Thanks a lot!

**Comment by joserbngithub** [comment]
As an end user, right clicking on the image (in windows) is already doing the image capture as a png! thanks @tobloef !

**Comment by brunotdantas** [comment]
So the solution would be a button to do the 'saving' ?

**Comment by brunotdantas** [comment]
@arendtio and @joserbngithub I created a mindmap to college but when saving this way it crops the image of the canvas, please check:  
![image](https://user-images.githubusercontent.com/19207320/41290397-27305492-6e23-11e8-8061-fc8391c8215b.png)

**Comment by arendtio** [comment]
@brunotdantas yes, in fact the canvas renders only your current viewport and when you save an image it saves only what is rendered. So @tobloef should bear that in mind for his rewrite. Maybe it would be possible to render the tree on some huge hidden canvas in order to create a picture.
